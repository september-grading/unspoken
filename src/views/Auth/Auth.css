.auth-container {
  display: flex;
  flex-direction: column;
  width: 500px;
  margin: auto;
}
.auth-tabs {
  display: flex;
  justify-content: space-evenly;
}

.auth-tabs h3 {
  color: red;
}

.auth-tabs h3.active {
  color: blue;
}
h3 {
  font-size: 2rem;
}

h3:hover {
  text-decoration: underline 2px black;
  width: fit-content;
}
