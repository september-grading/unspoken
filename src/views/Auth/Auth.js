import React, { useState } from 'react';
import { signInUser, signUpUser } from '../../services/users';
import AuthForm from '../../components/AuthForm/AuthForm';
import classNames from 'classnames';
import { useHistory } from 'react-router-dom';
import './Auth.css';

export default function Auth({ setCurrentUser }) {
  const [type, setType] = useState('signin');
  const [errorMessage, setErrorMessage] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [username, setUsername] = useState('');
  const history = useHistory();

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const resp =
        type === 'signin'
          ? await signInUser(email, password)
          : await signUpUser(username, email, password);
      setCurrentUser(resp);
      history.push('/posts');
    } catch (e) {
      setErrorMessage('Something went wrong. Please try again.');
    }
  };
  return (
    <div className="auth-container">
      <div className="auth-tabs">
        <h3
          onClick={() => {
            setType('signin');
          }}
          className={classNames({ active: type === 'signin' })}
        >
          Sign In
        </h3>

        <h3
          onClick={() => {
            setType('signup');
          }}
          className={classNames({ active: type === 'signup' })}
        >
          Sign Up
        </h3>
      </div>
      <AuthForm
        username={username}
        setUsername={setUsername}
        errorMessage={errorMessage}
        email={email}
        setEmail={setEmail}
        password={password}
        setPassword={setPassword}
        handleSubmit={handleSubmit}
        type={type}
      />
    </div>
  );
}
