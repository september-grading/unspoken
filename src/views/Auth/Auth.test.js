import { MemoryRouter } from 'react-router-dom';
import Auth from '../Auth/Auth';
import { render, screen } from '@testing-library/react';

test('should show signin', async () => {
  const { container } = render(
    <MemoryRouter>
      <Auth />
    </MemoryRouter>
  );
  await screen.findByText('Sign In');
  expect(container).toMatchSnapshot();
});
