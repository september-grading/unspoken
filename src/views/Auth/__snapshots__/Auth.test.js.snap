// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show signin 1`] = `
<div>
  <div>
    <span
      class="auth-tabs"
    >
      <h3
        class="active"
      >
        Sign In
      </h3>
      <h3
        class=""
      >
        Sign Up
      </h3>
    </span>
    <form
      class="auth-form"
    >
      
      <label>
        Email: 
      </label>
      <input
        type="email"
        value=""
      />
      <label>
        Password: 
      </label>
      <input
        type="password"
        value=""
      />
      <input
        class="submit-btn"
        type="submit"
      />
    </form>
  </div>
</div>
`;
